### PR TITLE
[Documentation:Markdown] Enhanced Markdown Documentation for Additional Features

### PR DESCRIPTION
**Pull request by NareshBiradar1**
Fixes https://github.com/Submitty/Submitty/issues/10072 , where the Markdown editor in Submitty lacked documentation for certain features, specifically tables. I have added comprehensive instructions on how to create tables in Markdown to enhance the documentation.

### CODE REVIEW

File: _docs/student/communication/markdown.md [hunk 1]
```diff
@@ -94,4 +94,19 @@ using markdown.
...
+    ```
+    To add a table, use three or more hyphens (---) to create each column’s header, and use pipes (|) to separate each column. For compatibility, you should also add a pipe on either end of the row.
+
+    | Syntax      | Description |
+    | ----------- | ----------- |
+    | Header      | Title       |
+    | Paragraph   | Text        |
+
+    ```
+    For more Information , [Click here](https://www.markdownguide.org/extended-syntax/#tables)
```
Comment (PythonCoderAS) [inline _docs/student/communication/markdown.md:110]:
```suggestion
    For more information, [click here](https://www.markdownguide.org/extended-syntax/#tables)
```

File: _docs/student/communication/markdown.md [hunk 2]
```diff
@@ -94,4 +94,19 @@ using markdown.
...
+
```
Comment (PythonCoderAS) [inline _docs/student/communication/markdown.md:111]:
```suggestion

    There are also several tools online that can aid with creating a markdown table, such as [this tool](https://tableconvert.com/markdown-generator).
```

File: _docs/student/communication/markdown.md [hunk 3]
```diff
@@ -94,4 +94,19 @@ using markdown.
    3. Item 3
    ```
 
+* **Table**
+
+    To add a table, use three or more hyphens `(---)` to create each column’s header, and use pipes `(|)` to separate each column. For compatibility, you should also add a pipe on either end of the row.
+
+    ```
```
Comment (PythonCoderAS) [inline _docs/student/communication/markdown.md:101]:
```suggestion
    ```markdown
```

File: _docs/student/communication/markdown.md [hunk 4]
```diff
@@ -94,4 +94,19 @@ using markdown.
...
+    To add a table, use three or more hyphens (---) to create each column’s header, and use pipes (|) to separate each column. For compatibility, you should also add a pipe on either end of the row.
```
Comment (PythonCoderAS) [inline _docs/student/communication/markdown.md:102]:
```suggestion
```

Comment (PythonCoderAS) [reply _docs/student/communication/markdown.md:102]: Duplicated text?

File: _docs/student/communication/markdown.md [hunk 5]
```diff
@@ -94,4 +94,19 @@ using markdown.
...
+
```
Comment (PythonCoderAS) [inline _docs/student/communication/markdown.md:103]:
```suggestion
```

File: _docs/student/communication/markdown.md [hunk 6]
```diff
@@ -94,4 +94,19 @@ using markdown.
...
+    | Syntax      | Description |
+    | ----------- | ----------- |
+    | Header      | Title       |
+    | Paragraph   | Text        |
+
```
Comment (PythonCoderAS) [inline _docs/student/communication/markdown.md:108]:
```suggestion
```

File: _docs/student/communication/markdown.md [hunk 7]
```diff
@@ -94,4 +94,19 @@ using markdown.
...
+    ```
```
Comment (PythonCoderAS) [inline _docs/student/communication/markdown.md:109]:
```suggestion
    ```
    The resulting table would look like this:
    | Syntax      | Description |
    | ----------- | ----------- |
    | Header      | Title       |
    | Paragraph   | Text        |
```

**Review by PythonCoderAS — CHANGES_REQUESTED** [review]
Just a few syntax corrections.

**Review by bmcutler — APPROVED** [review]

**Pull request merged**